selfdrive/car/gm/gmcan.py

```diff
@@ -81,7 +81,9 @@ def create_friction_brake_command(packer, bus, apply_brake, idx, enabled, near_s
 
   if apply_brake > 0:
     mode = 0xa
-    if at_full_stop:
+    if pre_enabled:
+      mode = 0xb
+    elif at_full_stop:
       mode = 0xd
```

Comment:
Can you try to send brake without making changes to the mode?

Reply:
I don't have an ASCM/stock ACC, but I could probably ask one of the other guys to record a route.

0xb by itself will allow pre-enable, then holds brakes for roughly half a second after user brake is released. After that, the system faults and releases the brake hold.

0xd by itself causes a fault while user brake is pressed IIRC.

Reply:
I'm surprised this isn't more widely reported then, are you sure it's not due to your car being non-stock? @Verylukyguy can you enable at a stop with your foot on the brake? What happens when you do so?

Reply:
> 

I can enable at a stop with my foot on the brake.  I can provide a test route, but not until later tonight.